src/Elastic.Apm.Azure.Storage/AzureBlobStorageDiagnosticListener.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Elastic.Apm.Api;
+using Elastic.Apm.DiagnosticListeners;
+using Elastic.Apm.Helpers;
+using Elastic.Apm.Logging;
+
+namespace Elastic.Apm.Azure.Storage
+{
+	internal static class AzureBlobStorage
+	{
+		internal const string SpanName = "AzureBlob";
+		internal const string SubType = "azureblob";
+	}
+
+	/// <summary>
+	/// Creates transactions and spans for Azure Blob Storage diagnostic events from Azure.Storage.Blobs
+	/// </summary>
+	internal class AzureBlobStorageDiagnosticListener : DiagnosticListenerBase
+	{
+		private readonly ConcurrentDictionary<string, IExecutionSegment> _processingSegments =
+			new ConcurrentDictionary<string, IExecutionSegment>();
+
+		public AzureBlobStorageDiagnosticListener(IApmAgent agent) : base(agent) { }
+
+		public override string Name { get; } = "Azure.Storage.Blobs";
+
+		protected override void HandleOnNext(KeyValuePair<string, object> kv)
+		{
+			Logger.Trace()?.Log("Called with key: `{DiagnosticEventKey}'", kv.Key);
+
+			if (string.IsNullOrEmpty(kv.Key))
+			{
+				Logger.Trace()?.Log($"Key is {(kv.Key == null ? "null" : "an empty string")} - exiting");
+				return;
+			}
+
+			switch (kv.Key)
+			{
+				case "BlobContainerClient.Create.Start":
+				case "PageBlobClient.Create.Start":
+					OnStart(kv, "Create");

```

Comment:
Any reason we use capital letter here (`"Create"`)? This will end up being the `action` for the span and I think we use lower case at other places. Definitely won't break anything, just wondering if this is intentional.

There are some other instances in the PR as well.

Reply:
Following [the convention used for AWS S3 span names](https://github.com/elastic/apm/blob/master/specs/agents/tracing-instrumentation-aws.md#s3-simple-storage-service) (the spec says operation name is camel case, but I think this should read pascal case based on the examples for S3, DynamoDB and the agent weekly conversations). Note that the span action is lowercase.

I don't really have an opinion about which might be best to use for this. Happy to follow up afterwards